File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

//import com.github.pagehelper.PageInfo;

import com.pinyougou.pojo.TbBrand;
import entity.PageInfo;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 *
 * @author Administrator
 */
public interface BrandService {

    /**
     * 查找所有并分页
     * @param pageNum
     * @param pageSize
     * @return PageInfo<TbBrand>
     */


    public PageInfo<TbBrand> findAll(Integer pageNum, Integer pageSize);

    public void add(TbBrand tbBrand);

    /**
     * update
     * @param tbBrand
     * @return
     */
    public void update(TbBrand tbBrand);

    /**
     * 查找一个
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    public void delete(Long[] ids);

    public  PageInfo<TbBrand> findAll(Integer pageNum, Integer pageSize, TbBrand tbBrand);

    List<Map<String, Object>> selectOptionList();
}

File: pinyougou-page-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    this.save = function (entity) {
        return $http.post('../brand/add.do', entity);
    };
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };
    this.deletea = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids)
    };
    this.search = function (currentPage, itemsPerPage, searchEntity) {
        return $http.post('../brand/search.do?page=' + currentPage + '&rows=' + itemsPerPage, searchEntity);
    };


    this.selectOptionList = function () {
        return $http.post('../brand/selectOptionList.do');
    }

});